addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -197,6 +197,48 @@ export class PosStore extends Reactive {
         this.computeProductPricelistCache();
     }
 
+    async deleteOrders(orders, serverIds = []) {
+        const ids = new Set();
+        for (const order of orders) {
+            if (order && (await this._onBeforeDeleteOrder(order))) {
+                if (Object.keys(order.last_order_preparation_change).length > 0) {
+                    await this.sendOrderInPreparationUpdateLastChange(order, true);
+                }
+
+                const cancelled = this.removeOrder(order, true);
+                if (!cancelled) {
+                    return false;
+                } else if (typeof order.id === "number") {
+                    ids.add(order.id);
+                }
+            }
+        }
+
+        if (serverIds.length > 0) {
+            for (const id of serverIds) {
+                if (typeof id !== "number") {
+                    continue;
+                }
+                ids.add(id);
+            }
+        }
```

Comment:
maybe instead of collecting the integer order ids in two different places, you can just do something like this:

```
const orderIds = [...orders.map(o => o.id), ...serverIds];
const validOrderIds = ordersIds.filter(oid => typeof oid === "number");
```

Then you give `validOrdersIds` to the `action_pos_order_cancel` rpc call.

Reply:
It was to avoid to looping in open orders again